File: server/src/models/relational/bangumi/common/SubjectCollection.ts
export interface SubjectCollection {
  /**
   * 想做
   * example: 608
   * format: int32
   */
  wish?: number;
  /**
   * 做过
   * example: 3010
   * format: int32
   */
  collect?: number;
  /**
   * 在做
   * example: 103
   * format: int32
   */
  doing?: number;
  /**
   * 搁置
   * example: 284
   * format: int32
   */
  on_hold?: number;
  /**
   * 抛弃
   * example: 86
   * format: int32
   */
  dropped?: number;
}

File: app/lib/models/bangumi/subject/SubjectCollection.dart
import 'dart:convert';

import 'package:built_value/built_value.dart';
import 'package:built_value/serializer.dart';
import 'package:munin/shared/utils/serializers.dart';

part 'SubjectCollection.g.dart';

@BuiltValue(wireName: 'Collection')
abstract class SubjectCollection
    implements Built<SubjectCollection, SubjectCollectionBuilder> {
  SubjectCollection._();

  factory SubjectCollection([updates(SubjectCollectionBuilder b)]) =
      _$SubjectCollection;

  @BuiltValueField(wireName: 'wish')
  int get wish;

  @BuiltValueField(wireName: 'collect')
  int get collect;

  @BuiltValueField(wireName: 'doing')
  int get doing;

  @BuiltValueField(wireName: 'on_hold')
  int get onHold;

  @BuiltValueField(wireName: 'dropped')
  int get dropped;

  String toJson() {
    return json
        .encode(serializers.serializeWith(SubjectCollection.serializer, this));
  }

  static SubjectCollection fromJson(String jsonString) {
    return serializers.deserializeWith(
        SubjectCollection.serializer, json.decode(jsonString));
  }

  static Serializer<SubjectCollection> get serializer =>
      _$subjectCollectionSerializer;
}
